### PR TITLE
SDIT-2556 More config and initial disclaimer flow added

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: server/@types/forms/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+declare module 'forms' {
+  export interface PrisonerSearchForm {
```
Comment (karenmillermoj) [inline server/@types/forms/index.d.ts:2]: This is really a placeholder - still wip

File: server/routes/index.test.ts [hunk 2]
```diff
@@ -23,18 +27,11 @@ afterEach(() => {
...
     return request(app)
       .get('/')
-      .expect('Content-Type', /html/)
       .expect(res => {
-        expect(res.text).toContain('This site is under construction...')
-        expect(auditService.logPageView).toHaveBeenCalledWith(Page.EXAMPLE_PAGE, {
-          who: user.username,
-          correlationId: expect.any(String),
-        })
+        expect(res.text).toContain('Redirecting to /disclaimer')
```
Comment (karenmillermoj) [inline server/routes/index.test.ts:34]: Wasn't sure if I needed a test for this

File: server/routes/search/search.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export default function routes(
+  router: Router,
+  auditService: AuditService,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  historicalPrisonerService: HistoricalPrisonerService,
+): Router {
+  const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
+  const post = (path: string, handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
+  // const controller = new SearchController(historicalPrisonerService)
```
Comment (karenmillermoj) [inline server/routes/search/search.ts:17]: wip

File: server/views/pages/disclaimer.njk [hunk 4]
```diff
@@ -26,29 +26,22 @@
           {{ govukCheckboxes({
```
Comment (karenmillermoj) [inline server/views/pages/disclaimer.njk:26]: More work may be needed to prettify the UI / marry up with original - but need all working first

File: integration_tests/e2e/disclaimer.cy.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import Page from '../pages/page'
+import Disclaimer from '../pages/disclaimer'
+
+context('Sign In', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn', { roles: ['ROLE_HPA_USER'] })
+  })
+
+  it('Will show the disclaimer text', () => {
```
Comment (karenmillermoj) [inline integration_tests/e2e/disclaimer.cy.ts:10]: Will also check original HPA for additional tests

Comment (karenmillermoj) [reply integration_tests/e2e/disclaimer.cy.ts:10]: Also need test to check for disclaimerConfirmed flag being set

File: server/middleware/validationMiddleware.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import HmppsError from '../interfaces/HmppsError'
+
+export type fieldErrors = {
+  [field: string | number | symbol]: string[] | undefined
+}
+
+export const customErrorOrderBuilder = (errorSummaryList: { href: string }[], order: string[]) =>
+  order.map(key => errorSummaryList.find(error => error.href === `#${key}`)).filter(Boolean)
+
+export const buildErrorSummaryList = (array: fieldErrors) => {
```
Comment (petergphillips) [inline server/middleware/validationMiddleware.ts:10]: don't think this is needed

File: server/routes/disclaimer/disclaimer.test.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+describe('GET /disclaimer', () => {
+  it('should render index/disclaimer page', () => {
+    auditService.logPageView.mockResolvedValue(null)
+
+    return request(app)
+      .get('/disclaimer')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Usage')
+        expect(auditService.logPageView).toHaveBeenCalledWith(Page.LOG_IN, {
```
Comment (petergphillips) [inline server/routes/disclaimer/disclaimer.test.ts:34]: probably need to review this at some stage, can't see the point of auditing that someone has signed into the service

File: server/routes/index.test.ts (hunk 2 above)
Comment (petergphillips) [reply server/routes/index.test.ts:34]: can't do any harm.

File: server/routes/search/search.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  get('/search', async (req, res, next) => {
+    logger.debug('search /')
+    if (req.session.disclaimerConfirmed) {
+      await auditService.logPageView(Page.SEARCH, { who: res.locals.user.username, correlationId: req.id })
+
+      res.render('pages/search')
+    } else {
+      res.render('pages/disclaimer')
```
Comment (petergphillips) [inline server/routes/search/search.ts:26]: not sure this is needed?  Shouldn't the disclaimer be an all routes middleware i.e. stopping them getting anywhere until confirmed?

File: server/views/pages/disclaimer.njk [hunk 9]
```diff
@@ -26,29 +26,22 @@
...
-              }
-            },
             items: [
               {
                 value: 'disclaimer',
                 text: 'I confirm that I understand'
               }
             ],
-            errorMessage: errors | findError('disclaimer')
+            errorMessage: validationErrors | findError('disclaimer')
```
Comment (petergphillips) [inline server/views/pages/disclaimer.njk:35]: might be good to go back to the original `errors`

**Review by petergphillips — APPROVED** [review]
LGTM

File: server/views/pages/disclaimer.njk (hunk 9 above)
Comment (karenmillermoj) [reply server/views/pages/disclaimer.njk:35]: yep - I was thinking that too

File: server/routes/disclaimer/disclaimer.test.ts (hunk 7 above)
Comment (karenmillermoj) [reply server/routes/disclaimer/disclaimer.test.ts:34]: good idea (review) - the current system audits the login so just replicated

File: server/routes/search/search.ts (hunk 8 above)
Comment (karenmillermoj) [reply server/routes/search/search.ts:26]: Updated/moved

**Pull request merged**